functions/payment.js

```diff
@@ -18,66 +17,30 @@ exports.getSessionId = functions.https.onCall(async (data, context) => {
     throw new functions.https.HttpsError("permission-denied", "Token invalid");
   }
 
-  if (!(data && data.venueId && data.eventId && data.returnUrl)) {
+  if (
+    !(data && data.venueId && data.eventId && data.userEmail && data.userId)
+  ) {
     throw new functions.https.HttpsError(
       "invalid-argument",
       "venue, event or return url data missing"
     );
   }
 
-  const getVenue = async () => {
-    const venueDoc = await firebase
-      .firestore()
-      .doc(`venues/${data.venueId}`)
-      .get();
-
-    return venueDoc.data();
-  };
-  const venue = await getVenue();
-
-  const getEvent = async () => {
-    return (
-      await firebase
-        .firestore()
-        .doc(`venues/${data.venueId}/events/${data.eventId}`)
-        .get()
-    ).data();
-  };
-  const event = await getEvent();
-
-  // @TODO: Check if venue and event entries exist
-  // @TODO: Check if purchase entry exist
-
-  // Do Stripe stuff
-  const session = await stripe.checkout.sessions.create({
-    payment_method_types: ["card"],
-    line_items: [
-      {
-        price_data: {
-          currency: "gbp",
-          product_data: {
-            name: `Ticket to ${event.name}@${venue.name}`,
-          },
-          unit_amount: event.price,
-        },
-        quantity: 1,
-      },
-    ],
-    mode: "payment",
-    success_url: `${data.returnUrl}?session_id={CHECKOUT_SESSION_ID}&redirectTo=payment_success`,
-    cancel_url: data.returnUrl,
+  const paymentIntent = await stripe.paymentIntents.create({
+    amount: data.price,
+    currency: "gbp",
```

Comment:
will it only be GBP

Reply:
for now it is, I'm going to ask the team this evening if they consider having multiple currencies